General Plugin Improvements

joemcgill opened the pull request:
- [x] Fix docs typos
- [x] Fix ESLint errors
- [x] Add escaping
- [x] Fix asset loading during dev
- [x] Fix PHP style

**Reviewer Note:** I didn't add CS rules to the plugin, but fixed ESLint based on HM standards and PHPCS based on WordPress-VIP standards. On the latter, I've intentionally ignored the `curl()` rule for now because it's only being used to check for API status, and switching to `wp_remote_get()` is changing the expected result.

joemcgill committed:
Improve inline docs

- Fixes spelling
- Adds punctuation
- Improves docblock formatting

joemcgill committed:
Fix ESLint errors

This fixes ESLint errors based on human made coding standards. Mostly includes whitespace fixes that cleans up reading the code.

joemcgill committed:
Add escaping to block output.

This adds escaping for attributes and URLs used when rendering the map block.

joemcgill committed:
Fix 1:1 ratio style

This fixes a typo in the CSS which kept the appropriate style from being applied to 1:1 ratio blocks.

joemcgill committed:
Commit built asset files

This commits build files and updates the hash used when registering scripts. It moves the preceding dot to the hash variable to make it easier to test during development by setting the has to an empty string.

joemcgill committed:
Meet WP-VIP PHP code standards

This fixes (or specifically ignores) PHPCS rules to pass WordPress-VIP coding standards.

joemcgill committed:
Remove unneccessary backslashes from global WP functions

If a function is called that doesn't exist in the current namespace, the function will be called in the global namespace. Since WordPress defines all its functions in the global space, we don't need to use backslashes in front of function names.

peterwilsoncc commented on embed-gutenberg-block-google-maps.php:
```diff
@@ -14,168 +14,171 @@
+	// Check status code of the API.
+	// phpcs:disable WordPress.WP.AlternativeFunctions
+	$ch = curl_init( $api_url );
+	curl_setopt( $ch, CURLOPT_HEADER, true );
```
Looks like this is the option that prevents the use of `WP_HTTP`, the remaining options are defaults/can be set via the API.

Part interesting fact/part desire to check if it's essential.

rmccue replied:
@peterwilsoncc It's never actually used, so irrelevant. You could replace this code with:

```php
$response = wp_remote_head( $api_url );
if ( is_wp_error( $response ) || wp_remote_retrieve_response_code( $response ) !== 200 ) {
    return;
}
```

peterwilsoncc commented on embed-gutenberg-block-google-maps.php:
```diff
@@ -14,168 +14,171 @@
+	}
+
+	// Set the appropriate CSS class names.
+	$classes = ( $interactive ) ? "wp-block-pantheon-google-map interactive ratio$aspectRatio" : 'wp-block-pantheon-google-map'; /* phpcs:ignore */
```
`/* phpcs:ignore */` ignores two lines, would be good to put it on line 117.5

Non-blocker

peterwilsoncc commented on embed-gutenberg-block-google-maps.php:
```diff
@@ -14,168 +14,171 @@
+	$classes = ( $interactive ) ? "wp-block-pantheon-google-map interactive ratio$aspectRatio" : 'wp-block-pantheon-google-map'; /* phpcs:ignore */
+
+	// Create the output.
+	$output = "<div class='$classes'><div class='map'>";
```
We'll need to add a couple of instances of escaping around here:

* `esc_attr( $classes )`
* `esc_url( $api_url )` x2

joemcgill replied:
Oh weird. Was handled in 0a25177c but must have gotten lost in another commit. Will redo.

peterwilsoncc reviewed (approved):
This looks good generally, the only real blocker is the lack of escaping of variables in the output. Once that is fixed I think it's good to go.

Reviewed with whitespace suppressed.

joemcgill committed:
Redo add escaping.

This was originally handled in 0a25177c but got lost in c7e38ab.

joemcgill committed:
Keep PHPCS from ignoring too many lines

From the docs:

> You can also ignore a single line using the phpcs:ignore comment. This comment will ignore the line that the comment is on, and the following line.

This moves the ignore rule from the line we want to ignore, to the line above the line we want to ignore, to keep the following line from being ignored, unintentionally.

joemcgill commented:
@peterwilsoncc I've fixed the critical issues and learned a new thing about PHPCS ignore rules. Cheers.
